wolfcrypt/src/asn.c

```diff
@@ -3883,9 +3597,23 @@ int wc_EncryptPKCS8Key(byte* key, word32 keySz, byte* out, word32 outSz,
     }
     if (ret == 0) {
         /* outer = SEQ [ pbe ] OCT encrypted_PKCS#8_key */
+        /* plus 2 for tag and length for pbe */
         outerLen = 2 + pbeLen;
-        outerLen += SetOctetString(keySz + padSz, out);
         outerLen += keySz + padSz;
+        if (out == NULL) {
+            /* plus 2 tags (for sequence and octet string). */
```

Comment:
// outerLen = length of PBE encoding + Octet string data
outerLen += 1 // Octet string tag
outerLen += SetLength(keySz + padSz, NULL); // Octet string Length
outSz = 1 + SetLength(outerLen, NULL) + outerLen;

Something like this would be clearer.